metricbeat/module/kubernetes/state_deployment/data.go

```diff
@@ -0,0 +1,64 @@
+package state_deployment
+
+import (
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/metricbeat/mb"
+	"github.com/elastic/beats/metricbeat/module/kubernetes/util"
+
+	dto "github.com/prometheus/client_model/go"
+)
+
+const (
+	// Nanocores conversion 10^9
+	nanocores = 1000000000
+)
+
+func eventMapping(families []*dto.MetricFamily) ([]common.MapStr, error) {
+	eventsMap := map[string]common.MapStr{}
+	for _, family := range families {
+		//fmt.Println(family)
+		for _, metric := range family.GetMetric() {
+			deployment := util.GetLabel(metric, "deployment")
+			if deployment == "" {
+				continue
+			}
+			event, ok := eventsMap[deployment]
+			if !ok {
+				event = common.MapStr{}
+				eventsMap[deployment] = event
+			}
+			switch family.GetName() {
+			case "kube_deployment_metadata_generation":
```

Comment:
I assume if you use prometheus module and get a mapStr back, you could use `schema` to do this mappings.